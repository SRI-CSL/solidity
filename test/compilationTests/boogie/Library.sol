pragma solidity ^0.4.23;

library MathLib {
    function add(uint256 a, uint256 b) internal pure returns (uint256) {
        return a + b;
    }
}

contract C {
    using MathLib for uint256;

    function someFunc() public pure returns (uint256) {
        uint256 x = 5;
        //return x.add(10); // Does not work yet
        return x;
    }
}

contract D {
    function otherFunc() public pure returns (uint256) {
        return MathLib.add(1, 2);
    }
}