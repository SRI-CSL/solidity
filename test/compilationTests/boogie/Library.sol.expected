Verifying Library::__verifier_main: OK
No errors found.
